lasagne/nonlinearities.py

```diff
@@ -286,6 +286,71 @@ def softplus(x):
     return theano.tensor.nnet.softplus(x)
 
 
+# Clipped Activation
+class ClippedActivation(object):
+    """Clipped Activation activation function
+    :math:`\\varphi(x,c,a) = \\min(a(x),c)`
+
+    This function clips the activation function. Usefull for non-saturating
+    functions such as the rectify to avoid exploding activations.
+
+    Parameters
+    ----------
+    x : float32
+        The activation (the summed, weighted input of a neuron).
+    clip : float32
+        The upper bound for the activation function.
+    activation : function
+        An activation function that maps x onto a float32
+
+    Methods
+    -------
+    __call__(x)
+        Apply the clipping to the to the activation `x`.
+
+    Examples
+    --------
+    In contrast to other activation functions in this module, this is
+    a class that needs to be instantiated to obtain a callable:
+
+    >>> from lasagne.layers import InputLayer, DenseLayer
+    >>> l_in = InputLayer((None, 100))
+    >>> from lasagne.nonlinearities import ClippedActivation, rectify
+    >>> custom_activation = ClippedActivation(clip=20, activation=rectify)
+    >>> l1 = DenseLayer(l_in, num_units=200, nonlinearity=custom_activation)
+
+    Alternatively, you can use the provided instance for clip=20,
+    activation=rectify
+
+    >>> from lasagne.nonlinearities import clipped_rectify
+    >>> l2 = DenseLayer(l_in, num_units=200, nonlinearity=clipped_rectify)
+    """
+
+    def __init__(self, clip=20, activation=rectify):
```

Comment:
If we call it `ClippedActivation`, maybe it should support all of the functionality of `T.clip()`. We could have two arguments `maximum` and `minimum`, with `minimum` defaulting to `None`. Depending on which bounds are given, `__call__` would use `T.minimum`, `T.maximum` or `T.clip`.